src/ai/defaultai_seafaring.cc

```diff
@@ -545,6 +585,27 @@ void DefaultAI::expedition_management(ShipObserver& so) {
 	so.waiting_for_command_ = false;
 }
 
+// this is called whenever a warship received a notification that requires
+// navigation decisions (these notifications are processes not in 'real time')
+void DefaultAI::warship_management(ShipObserver& so) {
+
+	const Time& gametime = game().get_gametime();
+
+	game().send_player_warship_command(*so.ship, Widelands::WarshipCommand::kSetCapacity, {0u});
+
+	for (auto i : portsites) {
+		if (i.guard_ship_needed) {
+			game().send_player_ship_set_destination(*so.ship, i.site->get_portdock());
+			i.guard_ship_needed = false;
+			i.ships_assigned = ++i.ships_assigned;
```

Comment:
This looks like it should be:

```suggestion
			++i.ships_assigned;
```

but I didn't check the wider context whether that's so, or the value should be assigned to some other variable.

Reply:
Or not... I never know the precedences... Maybe some parentheses are needed?

Reply:
omg you are right. what a stupid mistake from me